### PR TITLE
doc: a guide about how to use api-testing to try tRPC

### PR DESCRIPTION
**Pull request by LinuxSuRen**
close https://github.com/trpc-group/trpc-go/issues/100

@WineChord gives me a very big help. Thanks for that.

### CODE REVIEW

**Comment by LinuxSuRen** [comment]
I have read the CLA Document and I hereby sign the CLA

File: docs/user_guide/testing.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# tRPC-Go Testing
```
Comment (WineChord) [inline docs/user_guide/testing.md:1]: Could please also update the corresponding Chinese version? Just using chatGPT to generate should be fine.

Comment (LinuxSuRen) [reply docs/user_guide/testing.md:1]: Sure.

File: docs/README.md [hunk 2]
```diff
@@ -27,6 +27,7 @@ English | [中文](README.zh_CN.md)
   - [High Performance Network Package tnet](/docs/user_guide/tnet.md)
   - [Track Running Status](/rpcz/README.md)
   - [Attachment (Big Binary Data) Transfer](/docs/user_guide/attachment.md)
+  - [Testing](/docs/user_guide/testing.md)
```
Comment (WineChord) [inline docs/README.md:30]: Also add https://github.com/trpc-group/trpc-go/blob/main/docs/README.zh_CN.md

**Review by WineChord — APPROVED** [review]
LGTM

**Comment by LinuxSuRen** [comment]
[v0.0.14](https://github.com/LinuxSuRen/api-testing/releases/tag/v0.0.14) was released.

File: docs/user_guide/testing.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+`http://localhost:8080`
+
+![image](https://github.com/trpc-group/trpc-go/assets/1450685/fa9c66fc-ec5b-4a70-9466-f6d923aac229)
+![image](https://github.com/trpc-group/trpc-go/assets/1450685/03347a59-51a4-43ed-aa44-7b0dec340b90)
+![image](https://github.com/trpc-group/trpc-go/assets/1450685/12d1c7b0-bffd-4a48-adc4-5ed4a156afef)
+![image](https://github.com/trpc-group/trpc-go/assets/1450685/0c264a5e-dec0-400a-a420-55af79dbedbd)
+
+## Read More
+[api-testing](https://github.com/linuxsuren/api-testing) has several backend storage, it will save the data as a regular file if you just want to have a try. But it also can save the data to an ORM database or git repository. Please feel free to read more detail from the official document.
```
Comment (WineChord) [inline docs/user_guide/testing.md:25]: @LinuxSuRen Hi, the testing part in trpc-go/docs should provide a high-level, general overview, without placing too much emphasis on specific tools. I believe the bulk of the document should explore broader testing concepts. The tools can be mentioned in the final section, with brief references to external repositories being adequate. 
Here's my overly simplified suggestion:

```suggestion
# tRPC-Go Testing

Interface testing refers to the process in which, after the service has successfully started, verification is carried out by sending packets.

## Tools

* [trpc-cli](TODO)
* [api-testing](https://github.com/linuxsuren/api-testing)
* [eab](TODO)
```

Please note that trpc-cli and eab are internal tools used within Tencent that are yet to be released.

Comment (LinuxSuRen) [reply docs/user_guide/testing.md:25]: Sounds reasonable.

**Review by sandyskies — APPROVED** [review]
LGTM

**Pull request merged**